index.js

```diff
@@ -91,9 +73,61 @@ server.get('/api/projects/:id' , (req , res) => {
     .catch(err => {
         res.status(500).json(err)
     })
-})    
+})      
+
+ //UPDATE a project 
+
+ router.put("/api/projects/:id", (req, res) => {
+    const id = req.params.id;
+    const updatedproject = req.body;
   
-// post a new action 
+    if (!req.body.name || !req.body.description) {
+      res.status(404).json({ error: "can not update" });
+    } else {
+      projectModel.update(id, updatedproject).then(response => {
+        if (!response) {
+          res.status(400).json(response);
+        } else {
+          res.status(200).json(response);
+        }
+      });
+    }
+  });
+
+// DELETE a project 
+
+server.delete('/api/projects/:id' , validateProjectId, (req, res) => {
+    const {id} = req.params;
+    projectModel.remove(id) 
+    .then(response => {
+        if(response){
+          res.status(204).json(response); 
+        } else {
+          res.status(404).json({ error: 'Project with id does not exist' });
+        }
+    })
+    .catch(err => {
+      res.status(500).json({ error: 'Error removing project' });
+    })
+  })   
+
+
+//ACTIONS ......
+
+// GET all the actions 
+
+server.get('/api/actions' , (req , res) => {
+    actionsModel.get()  
+    .then(response => {
+        console.log(response)
+        res.status(200).json(response)
+    })
+    .catch(err => {
+        res.status(500).json(err)
+    })
+})   
+
+// CREATE a new action 
 
 server.post('/api/actions/:id' , (req , res) => {
```

Comment:
MVP states that you must validate the project id and error out if it's incorrect when creating a new action.